### PR TITLE
Inspection and resolve fixes

### PR DESCRIPTION
**Pull request by mbolotov**
1. Member access inspection 
2. Fixes for a huge number of problems found by this and other inspections

### CODE REVIEW

**Comment by mbolotov** [comment]
@SerCeMan 
when you will be making the next release of plugin, could you also mention [tetu.io](tetu.io) who made this development possible

**Comment by SerCeMan** [comment]
@mbolotov will do 👍

File: src/main/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -207,6 +207,14 @@
                          level="WEAK WARNING"
                          implementationClass="me.serce.solidity.ide.inspections.VisibilityModifierInspection"/>
 
+        <localInspection language="Solidity" groupName="Solidity"
+                                 displayName="Member is not defined"
+                                 enabledByDefault="true"
+                                 level="ERROR"
```
Comment (SerCeMan) [inline src/main/resources/META-INF/plugin.xml:213]: I did a few rounds of testing on various repositories, and it seems that due the current incomplete resolution strategy, the inspection produces a large number of false positives. Too large to be enabled by default. 

For example, the inspection currently fails for types imported through `import as` or  the following code results in an error:
```solidity
// SPDX-License-Identifier: GPL-3.0

pragma solidity >=0.7.0 <0.9.0;

contract B {
    type ShortString is bytes32;
    
    bytes32 private constant V = 0x00000000000000000000000000000000000000000000000000000000000000FF;

    function a() public pure returns (ShortString) {
        return ShortString.wrap(V);
    }
}
```

What do you think of either only scoping the inspection to the cases where the receiver was actually resolved, and it's not unknown? Or, alternatively, making it a `WEAK WARNING`?

File: src/main/resources/META-INF/plugin.xml [hunk 2]
```diff
@@ -207,6 +207,14 @@
...
+                                 implementationClass="me.serce.solidity.ide.inspections.ValidateMemberAccessInspection"/>
+
+
+
```
Comment (SerCeMan) [inline src/main/resources/META-INF/plugin.xml:217]:
```suggestion
```

nit

File: src/main/kotlin/me/serce/solidity/lang/types/types.kt [hunk 3]
```diff
@@ -264,20 +273,43 @@ data class SolStructVariableDeclaration(
...
 
+  override fun getName(): String? = ref.name
+
+  override fun parseType(): SolType = this
+
+  override fun resolveElement(): SolNamedElement? = ref
+
+  override fun getPossibleUsage(contextType: ContextType): Usage? = null
+
+
```
Comment (SerCeMan) [inline src/main/kotlin/me/serce/solidity/lang/types/types.kt:304]:
```suggestion
```

nit

**Review by SerCeMan — COMMENTED** [review]
Thank you for the PR! I have a suggestion for the inspection's level.

File: src/main/resources/META-INF/plugin.xml (hunk 1 above)
Comment (mbolotov) [reply src/main/resources/META-INF/plugin.xml]: hmm, it seems my yesterday reply was lost.
regarding this I did the following:
1. Set `WEAK WARNING` as the default level for inspection 
2. Update inspection to now report an error when the receiver was not resolved
3. Fixed the case with resolving a user defined value type

**Review by SerCeMan — APPROVED** [review]
Thank you 🙌

**Pull request merged**